### PR TITLE
Fire alarm sound

### PR DESCRIPTION
**Pull request by west3436**
## What this does

Plays a siren sound when a fire alarm is triggered.

## Why it's good

Adds for more \~atmosphere\~ when caught in a room on fire.

## Changelog

:cl:
 * rscadd: Fire alarms will now make a sound when they are triggered.

### CODE REVIEW

**Comment by pedr69** [comment]
it's a one time thing right?
or will a shartbomb result in everyone poking a screwdriver into their ear to stop the noise

**Comment by west3436** [comment]
> it's a one time thing right? or will a shartbomb result in everyone poking a screwdriver into their ear to stop the noise

Yeah it's just one-time.

**Comment by west3436** [comment]
> does the alarm sound still play when someone manually pulls the fire alarm like the fire alarms in school

Yes

**Comment by Hinaichigo** [comment]
Nice. This tangentially makes me think that the emergency shuttle red lighting overlay should also be replaced by actual alarms with red lightbulbs and repeating klaxon sounds. It would require mapping though...

**Comment by adacovsk** [comment]
I got permanent tinnitus from a fire alarm at my workplace. Please ensure it's one of those nice alarm sounds, and not the ear-drum piercing noise

**Comment by west3436** [comment]
> I got permanent tinnitus from a fire alarm at my workplace. Please ensure it's one of those nice alarm sounds, and not the ear-drum piercing noise

It's the weather alert sound used on snaxi. Also open to stealing that muffled alarm sound from Star Wars.

**Comment by gurfan** [comment]
I'm not a fan of the sound used, it doesn't really fit a fire alarm very well. It's more of a distant siren than the harsh noise you would expect from a fire alarm.

Also this is extremely spammable, expect clowns to toggle fire alarms repeatedly to rape everyone's ears. I think it would be better if the sound was a quieter, more ambient thing played by any active fire alarm rather than something that plays every time you trigger it.

**Comment by WindowsErrors** [comment]
[here mate try this](https://www.youtube.com/playlist?list=PLYNzgrWx34qqmHMnTjjDUJH8FiGQU9kjz)

**Comment by west3436** [comment]
> I'm not a fan of the sound used, it doesn't really fit a fire alarm very well. It's more of a distant siren than the harsh noise you would expect from a fire alarm.
> 
> Also this is extremely spammable, expect clowns to toggle fire alarms repeatedly to rape everyone's ears. I think it would be better if the sound was a quieter, more ambient thing played by any active fire alarm rather than something that plays every time you trigger it.

I'll look at other sounds to see if there's a more suitable one. As for the spam comment, there are lots of things that do this already (comms console code green vs blue, AI VOX announcements, shuttle spam, etc)

**Comment by Hinaichigo** [comment]
Is the sound global or does it only play in the proximity of the alarm?

**Comment by west3436** [comment]
> Is the sound global or does it only play in the proximity of the alarm?

Only in the room where it is triggered

**Comment by pedr69** [comment]
>emag fire alams
>they play [loud [redacted]](https://www.youtube.com/watch?v=IFOvTWSYCbc) when triggered

**Comment by Kurfursten** [comment]
I also don't like the sound used. If people associate that sound with fire alarms being pulled, it won't register to them as a weather warning on Snaxi

**Comment by west3436** [comment]
> I also don't like the sound used. If people associate that sound with fire alarms being pulled, it won't register to them as a weather warning on Snaxi

Any suggestions on an existing ogg to reference or should I just convert an IRL (jarring) fire alarm sound into an ogg and use it?

**Comment by Kurfursten** [comment]
I think that this is a nice sound because it is lower pitch than most alarms, meaning it is less annoying. I would pull JUST seconds 13-15. It tapers off nicely and you really only need about 2 seconds to get the point. https://www.youtube.com/watch?v=qrNZrr9lD7k

I believe if you use a lower pitch bell and keep the volume and length manageable, it won't irritate the ears.

**Comment by west3436** [comment]
Draft until I get the sound updated later today

**Comment by Kurfursten** [comment]
This is cut entirely wrong. You didn't get the taper at the end correct and it's way too long.

**Comment by west3436** [comment]
> This is cut entirely wrong. You didn't get the taper at the end correct and it's way too long.

By design, I wanted to keep it long since it doesn't loop and only plays once

**Comment by Kurfursten** [comment]
It does not play once, it plays once every time it alarms. Meaning you can pull the fire alarm over and over and over again. If I click at the reasonable rate of 10 times per second, I can queue up 5 of these sound effects every second. Meaning over the 13 second period, I can start up 65 new 13-second clips. Don't be foolish with this sound size, 2 seconds is plenty.

File: code/game/machinery/alarm.dm [hunk 1]
```diff
@@ -1264,7 +1265,9 @@ FIRE ALARM
 		shelter = TRUE
 		update_icon()
 		return
-
+	if(istype(W,/obj/item/weapon/card/emag))
```
Comment (Kurfursten) [inline code/game/machinery/alarm.dm]: instead of doing this, just call ..() at the start of attackby -- that's the whole point of the emaggable flag

**Comment by pedr69** [comment]
reminder you can embed mp4's in the PR body
`ffmpeg -i filename.ogg output.mp4`
then drag the file into the web text editor

File: code/game/machinery/alarm.dm (hunk 1 above)
Comment (west3436) [reply code/game/machinery/alarm.dm]: thanks, implemented in latest commit

**Comment by Kurfursten** [comment]
Even with the shorter sfx, we may need to add a cooldown to fire alarms. However, I think for now this is good.,

**Pull request merged**